quickjs.c

```diff
@@ -6673,26 +6675,43 @@ static void build_backtrace(JSContext *ctx, JSValue error_val, JSValue filter_fu
     JSRuntime *rt;
     JSCallSiteData csd[64];
     uint32_t i;
+    double d;
     int stack_trace_limit;
 
+    rt = ctx->rt;
+    if (rt->in_build_stack_trace)
+        return;
+    rt->in_build_stack_trace = true;
+
+    // Save exception because conversion to double may fail.
+    saved_exception = JS_GetException(ctx);
+
+    // Extract stack trace limit.
+    JS_ToFloat64(ctx, &d, ctx->error_stack_trace_limit);
+    if (isnan(d) || d < 0.0 || (isinf(d) && d < 0.0))
+        stack_trace_limit = 0;
+    else if (isinf(d) || d > (double)INT32_MAX)
```

Comment:
You don't have to cast to double (but it doesn't hurt either, it's just a no-op)

The `isinf(d) && d < 0.0` two lines up looks superfluous. You check `d < 0.0` first and `-Infinity < 0.0`

Reply:
Right. Does the same apply to checking for positive infinity? It will be larger than INT32_MAX I guess.

Reply:
Updated!

Reply:
> Right. Does the same apply to checking for positive infinity? It will be larger than INT32_MAX I guess.

Good point. By an infinite amount, according to mathematicians.